lib/pacemaker/pcmk_sched_probes.c

```diff
@@ -436,52 +429,59 @@ order_then_probes(pe_working_set_t * data_set)
             g_list_free(actions);
         }
 
-        if(start == NULL) {
+        if (start == NULL) {
             crm_err("No start action for %s", rsc->id);
             continue;
         }
 
         probes = pe__resource_actions(rsc, NULL, RSC_STATUS, FALSE);
 
-        for (actions = start->actions_before; actions != NULL; actions = actions->next) {
+        for (actions = start->actions_before; actions != NULL;
+             actions = actions->next) {
+
             pe_action_wrapper_t *before = (pe_action_wrapper_t *) actions->data;
 
-            GList *pIter = NULL;
             pe_action_t *first = before->action;
             pe_resource_t *first_rsc = first->rsc;
 
-            if(first->required_runnable_before) {
-                GList *clone_actions = NULL;
-                for (clone_actions = first->actions_before; clone_actions != NULL; clone_actions = clone_actions->next) {
+            if (first->required_runnable_before) {
+                for (GList *clone_actions = first->actions_before;
+                     clone_actions != NULL;
+                     clone_actions = clone_actions->next) {
+
                     before = (pe_action_wrapper_t *) clone_actions->data;
 
-                    crm_trace("Testing %s -> %s (%p) for %s", first->uuid, before->action->uuid, before->action->rsc, start->uuid);
+                    crm_trace("Testing '%s then %s' for %s",
+                              first->uuid, before->action->uuid, start->uuid);
 
-                    CRM_ASSERT(before->action->rsc);
+                    CRM_ASSERT(before->action->rsc != NULL);
                     first_rsc = before->action->rsc;
                     break;
                 }
 
-            } else if(!pcmk__str_eq(first->task, RSC_START, pcmk__str_casei)) {
+            } else if (!pcmk__str_eq(first->task, RSC_START, pcmk__str_none)) {
                 crm_trace("Not a start op %s for %s", first->uuid, start->uuid);
             }
 
-            if(first_rsc == NULL) {
+            if (first_rsc == NULL) {
                 continue;
 
-            } else if(uber_parent(first_rsc) == uber_parent(start->rsc)) {
+            } else if (uber_parent(first_rsc) == uber_parent(start->rsc)) {
                 crm_trace("Same parent %s for %s", first_rsc->id, start->uuid);
                 continue;
 
-            } else if(FALSE && pe_rsc_is_clone(uber_parent(first_rsc)) == FALSE) {
+            } else if (!pe_rsc_is_clone(uber_parent(first_rsc))) {
                 crm_trace("Not a clone %s for %s", first_rsc->id, start->uuid);
```

Comment:
This is a weird block, but previously it would never run and now it at least potentially might.  Is that what you intended?

Reply:
Well this is all in an #if 0 block, so I thought it was kind of redundant to disable this test in particular :)